src/terminal_builder.rs

```diff
@@ -0,0 +1,386 @@
+#[allow(unused_imports)]
+use std::io::{self, stderr, stdout, Stderr, Stdout, Write};
+
+use crate::{
+    backend::{self},
+    Terminal,
+};
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum RawMode {
+    #[default]
+    Enabled,
+    Disabled,
+}
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum AlternateScreenMode {
+    #[default]
+    EnterAlternateScreen,
+    MainScreen,
+}
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum MouseCapture {
+    Captured,
+    #[default]
+    NotCaptured,
+}
+
+#[derive(Debug)]
+pub struct TerminalBuilder;
+
+impl TerminalBuilder {
+    #[cfg(feature = "crossterm")]
+    pub fn crossterm_on_stdout() -> crossterm::TerminalBuilder<Stdout> {
+        Self::crossterm(stdout())
+    }
```

Comment:
These methods could also just hang directly off the module (either `terminal_builder` or `terminal` instead of needing an unnecessary struct. I wouldn't mind feedback on that idea. It would make startup something like:
```rust
let terminal = ratatui::terminal::crossterm_on_stdout().build()?;
// or
let terminal = ratatui::terminal_builder::crossterm_on_stdout().build()?;
// or 
let terminal = ratatui::crossterm_on_stdout().build()?;
```